app/src/main/java/org/dhis2/data/service/SyncPresenterImpl.kt

```diff
@@ -452,8 +452,12 @@ class SyncPresenterImpl(
             .byUid().eq(uid)
             .byAggregatedSyncState().notIn(State.SYNCED, State.RELATIONSHIP)
             .blockingGet().isEmpty()
+        val teiEventsOk = d2.eventModule().events()
+            .byEnrollmentUid().eq(uid)
+            .byAggregatedSyncState().notIn(State.SYNCED)
+            .blockingGet().isEmpty()
 
-        if (teiOk) {
+        if (teiOk && teiEventsOk) {
```

Comment:
I think syncPresenterImpl has unit tests. Perhaps we could cover it. thoughts?

Reply:
Yes, you are right!

Reply:
@JaimeToca Tests added!